vision_agent_tools/models/sam2.py

```diff
@@ -79,19 +83,78 @@ def predict_image(
-            scores = scores[sorted_ind]
-            logits = logits[sorted_ind]
-
         return masks, scores, logits
 
+    @torch.inference_mode()
+    def predict_video(
+        self,
+        video: VideoNumpy,
+        input_box: Optional[np.ndarray] = None,
+        input_points: Optional[np.ndarray] = None,
+        input_label: Optional[np.ndarray] = None,
+    ):
+        """
+        Process the input video with the SAM2 video predictor using the given prompts.
+
+        Args:
+            video: Input video to be processed.
+            input_box (Optional[np.ndarray]): Coordinates for boxes.
+            input_points (Optional[np.ndarray]): Coordinates for points.
+            input_label (Optional[np.ndarray]): Labels for the points.
+
+        """
+        if input_box is None and input_points is None and input_label is None:
+            raise ValueError(
+                "Either 'input_box' or 'input_points' and 'input_label' must be provided."
+            )
+
+        torch_dtype = (
+            torch.bfloat16 if self.model_config.device == Device.GPU else torch.float16
+        )
+
+        inference_state = self.video_model.init_state(video)
+
+        with torch.autocast(device_type=self.model_config.device, dtype=torch_dtype):
+            if (
+                self.model_config.device == Device.GPU
+                and torch.cuda.get_device_properties(0).major >= 8
+            ):
+                torch.backends.cuda.matmul.allow_tf32 = True
+                torch.backends.cudnn.allow_tf32 = True
+
+            for frame_idx in range(video.shape[0]):
+                np_image = np.array(video[frame_idx].convert("RGB"), dtype=np.uint8)
+                self.image_model.set_image(np_image)
+
+                self.video_model.add_new_points_or_box(
+                    inference_state=inference_state,
+                    frame_idx=frame_idx,
+                    obj_id=0,
+                    box=input_box,
+                    points=input_points,
+                    labels=input_label,
+                )
+
+            all_masks, all_scores, all_logits = [], [], []
+            for frame_idx, obj_ids, mask_logits in self.video_model.propagate_in_video(
+                inference_state
+            ):
+                all_masks.append((mask_logits > 0.0).astype(np.uint8))
+                all_scores.append(
+                    np.ones((len(obj_ids),))
+                )  # Assuming scores are binary masks
+                all_logits.append(mask_logits)
+
+        self.video_model.reset_state(inference_state)
+
+        return np.array(all_masks), np.array(all_scores), np.array(all_logits)
+
     @validate_call(config={"arbitrary_types_allowed": True})
     @torch.inference_mode()
     def __call__(
         self,
-        image: Image.Image | None = None,
-        video: VideoNumpy | None = None,
+        image: Optional[Image.Image] = None,
+        video: Optional[VideoNumpy] = None,
```

Comment:
Please don't use Optional, since we are trying to move towards the latest python typing changes.

Reply:
Ok!! Will remove from all the codes I am using